pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -235,7 +235,12 @@ func (m *ManagerImpl) PluginConnected(resourceName string, p plugin.DevicePlugin
 
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
-	m.endpoints[resourceName] = endpointInfo{e, options}
+	newEndpoint := endpointInfo{e, options}
+	if ep, exists := m.endpoints[resourceName]; exists {
+		klog.V(2).InfoS("Endpoint is taken over", "resourceName", resourceName, "prevEndpoint", ep, "newEndpoint", newEndpoint)
+		ep.e.setStopTime(time.Now())
+	}
+	m.endpoints[resourceName] = newEndpoint
```

Comment:
One thing that I don't see called out in this PR at the moment is whether plugins should or should not use the same socket path for each instance.

@linxiulei: can you clarify?

Reply:
I think they shouldn't because they can't listen on the same unix socket. This is possible for TCP sockets in `SO_REUSEADDR` mode, but `SO_REUSEADDR` doesn't have any effect on unix sockets as far as I know.

Reply:
This is up to plugin authors/owners. If they wish to have HA for their plugin, they should use unique socket paths in  because HA requires starting new Pod before stopping the old Pod. In other words, with the rolling-update, if `maxSurge` is 0, the socket path could be identical as there is only one Pod running on a node during an update.

Reply:
Agreed. This is something that we need to tell developers about. Making this code change and then not having any device plugin make use of it would be pointless.